### PR TITLE
Updated references master to main

### PR DESCRIPTION
**Pull request by axel-lauer**
This PR updates all references to "master" e.g. in the documentation and readme files (master branch) to "main" (the new GitHub standard, see https://github.com/github/renaming). See also corresponding PR in ESMValTool: https://github.com/ESMValGroup/ESMValTool/pull/2172.

Closes https://github.com/ESMValGroup/ESMValCore/issues/1031

***

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [ ] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

### CODE REVIEW

**Comment by valeriupredoi** [comment]
cheers @axel-lauer let me check and commit if need be the tests conf files :+1:

**Comment by axel-lauer** [comment]
@ESMValGroup/esmvaltool-coreteam are we now theoretically ready to rename master to main and then merge this PR and https://github.com/ESMValGroup/ESMValTool/pull/2172?

**Comment by valeriupredoi** [comment]
Axel, it's a go from my side, mate! we should probably add a test (GA or Circle) detect any instance of creation of a branch called `master` after we've switched to `main` and inspect/delete that, what do you think @bouweandela

**Review by valeriupredoi — APPROVED** [review]
fine by me, cheers much @axel-lauer  -> @bouweandela could you give it a look too when you gots some review time pls mate :beer:

File: .github/workflows/action-install-from-conda.yml [hunk 1]
```diff
@@ -13,19 +13,19 @@
 
 name: Conda Base Install
 
-# runs on a push on master and at the end of every day
+# runs on a push on maain and at the end of every day
```
Comment (bouweandela) [inline .github/workflows/action-install-from-conda.yml:16]:
```suggestion
# runs on a push on main and at the end of every day
```

**Comment by valeriupredoi** [comment]
Come on @bouweandela -> `maain` sounds quite Dutch, surprised you didn't like it :rofl:

**Comment by bouweandela** [comment]
The tests are failing, but it looks like that was solved in #1140, so merging `master` into this branch will probably solve that?

**Comment by valeriupredoi** [comment]
on it now @bouweandela - you reckon it'll be ready for merge after? I think it's all OK

**Comment by valeriupredoi** [comment]
checks have passed, gonna merge, cheers @axel-lauer and @bouweandela :beer: Shall we delete `master` once and for good or keep it just in queso

**Pull request merged**